sdk/lib/crt/string/tcsnlen.h

```diff
@@ -4,11 +4,11 @@
 
 size_t __cdecl _tcsnlen(const _TCHAR * str, size_t count)
 {
- const _TCHAR * s;
+ const _TCHAR * s = str;
 
- if(str == 0) return 0;
+ if(s == 0) return 0;
 
- for(s = str; *s && count; ++ s, -- count);
+ while ( *s && count ) { ++ s, -- count; };
```

Comment:
What warning does this fix? (to me, the `s` pointer looks to be of the same type as `str` so the original code should have compiled without warnings...... )